### PR TITLE
fix(section_model): fix missing role to update the access (request)

### LINKED ISSUE #11091: Community: Request to join took effect but the checkbox value was not persisted
# Bug Report

## Description

Found during #10909 

## Steps to reproduce

1. Start with open community
2. Edit community on Admin account
3. Toggle request to join and save
4. Open edit community on Admin again and observe the value has not been saved
5. Open edit community on Owner node and observe the value has not been saved
6. Attempt to join community as member and found request to join setting has been applied

This may lead to the owner assuming that request to join is/is not applied.

### Additional Information

- Status desktop version: Mac
- Operating System: https://ci.status.im/job/status-desktop/job/branches/job/macos/job/aarch64/job/package/job/PR-10909/25/

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #11091

[req-to-join.webm](https://github.com/status-im/status-desktop/assets/11926403/d5ee5f55-1109-417f-a920-6a29c1e35097)

### CODE REVIEW

File: src/app/modules/shared_models/section_model.nim [hunk 1]
```diff
@@ -292,6 +290,7 @@ QtObject:
       ModelRole.CanJoin.int,
       ModelRole.Joined.int,
       ModelRole.Spectated.int,
+      ModelRole.Access.int,
```
Comment (jrainville) [inline src/app/modules/shared_models/section_model.nim:293]: Did we have a way to just say "all ModelRoles"? That would be way better than adding each one manually. It can cause issues like this.

Comment (osmaczko) [reply src/app/modules/shared_models/section_model.nim:293]: I am not aware of any.

**Review by osmaczko — APPROVED** [review]

File: src/app/modules/shared_models/section_model.nim (hunk 1 above)
Comment (caybro) [reply src/app/modules/shared_models/section_model.nim:293]: Should be possible, just by passing an empty array, eg `@[]`; this is further passed on to this nimqml code:
```nim
proc dataChanged*(self: QAbstractItemModel,
                 topLeft: QModelIndex,
                 bottomRight: QModelIndex,
                 roles: openArray[int]) =
  ## Notify the view that the model data changed
  debugMsg("QAbstractItemModel", "dataChanged")
  var copy: seq[cint]
  for i in roles:
    copy.add(i.cint)
  dos_qabstractitemmodel_dataChanged(self.vptr.DosQAbstractItemModel, topLeft.vptr,
                                     bottomRight.vptr, copy[0].addr, copy.len.cint)
```

which then ends up calling the C++ method:
```cpp
void dos_qabstractitemmodel_dataChanged(::DosQAbstractItemModel *vptr,
                                        const ::DosQModelIndex *topLeftIndex,
                                        const ::DosQModelIndex *bottomRightIndex,
                                        int *rolesArrayPtr,
                                        int rolesArrayLength)
{
    auto object = static_cast<QObject *>(vptr);
    auto model = dynamic_cast<DOS::DosIQAbstractItemModelImpl *>(object);
    auto topLeft = static_cast<const QModelIndex *>(topLeftIndex);
    auto bottomRight = static_cast<const QModelIndex *>(bottomRightIndex);
    auto roles = QVector<int>(rolesArrayPtr, rolesArrayPtr + rolesArrayLength);
    model->publicDataChanged(*topLeft, *bottomRight, roles);
}
```
where the Qt method signature is:
```cpp
void dataChanged(const QModelIndex &topLeft, const QModelIndex &bottomRight, const QVector<int> &roles = QVector<int>());
```

TLDR, passing an empty `roles` vector means _all_ roles

Comment (caybro) [reply src/app/modules/shared_models/section_model.nim:293]: Yeah looks like it, should be easy to fix

Comment (caybro) [reply src/app/modules/shared_models/section_model.nim:293]: @jrainville can you quickly try this patch? :)

```diff
--- vendor/DOtherSide/lib/src/DOtherSide.cpp
+++ vendor/DOtherSide/lib/src/DOtherSide.cpp
@@ -1301,7 +1301,10 @@ void dos_qabstractitemmodel_dataChanged(::DosQAbstractItemModel *vptr,
     auto model = dynamic_cast<DOS::DosIQAbstractItemModelImpl *>(object);
     auto topLeft = static_cast<const QModelIndex *>(topLeftIndex);
     auto bottomRight = static_cast<const QModelIndex *>(bottomRightIndex);
-    auto roles = QVector<int>(rolesArrayPtr, rolesArrayPtr + rolesArrayLength);
+    QVector<int> roles;
+    if (rolesArrayPtr && rolesArrayLength > 0)
+        roles = QVector<int>(rolesArrayPtr, rolesArrayPtr + rolesArrayLength);
+
     model->publicDataChanged(*topLeft, *bottomRight, roles);
 }
```

Comment (jrainville) [reply src/app/modules/shared_models/section_model.nim:293]: Sorry, it doesn't seem to work either. I'm not sure what's the issue

Comment (caybro) [reply src/app/modules/shared_models/section_model.nim:293]: OK, can you please create an issue to track and assign to me? I have some local code to fix but let's deal with it separately

**Review by caybro — APPROVED** [review]
LGTM

File: src/app/modules/shared_models/section_model.nim (hunk 1 above)
Comment (jrainville) [reply src/app/modules/shared_models/section_model.nim:293]: Done here: https://github.com/status-im/status-desktop/issues/11830

Comment (caybro) [reply src/app/modules/shared_models/section_model.nim:293]: Thanks :)
